### PR TITLE
Add `-relative` option to ALESymbolSearch

### PR DESCRIPTION
**Pull request by chaucerbao**
I added the `-relative` option to ALESymbolSearch, and the associated test.

However, I only use TSServer, which doesn't seem to be implemented for **ALESymbolSearch**, so I'm unable to actually test the `-relative` option IRL. I kept changes minimal, and the existing tests pass. But I would certainly prefer a quick real-world test to confirm.

### CODE REVIEW

File: autoload/ale/symbol.vim [hunk 1]
```diff
@@ -76,34 +76,46 @@ function! s:OnReady(linter, lsp_details, query, ...) abort
...
     let l:buffer = bufnr('')
+    let l:query = []
+    let l:options = {}
+
+    " Separate arguments from flags
+    for l:arg in split(a:args)
+        if l:arg is? '-relative'
+            let l:options.use_relative_paths = 1
+        else
+            call add(l:query, l:arg)
```
[the remark this answers is not in the document]
Comment (chaucerbao) [reply autoload/ale/symbol.vim]: I separated the argument parsing into its own function. It only parses command switches before the query string. And I throw the error message if the query string is empty.

File: autoload/ale/symbol.vim [hunk 2]
```diff
@@ -76,34 +76,46 @@ function! s:OnReady(linter, lsp_details, query, ...) abort
...
-        \   function('s:OnReady', [a:linter, l:lsp_details, a:query]),
+        \   function('s:OnReady', [a:linter, l:lsp_details, a:query, a:options]),
         \)
     endif
 endfunction
 
-function! ale#symbol#Search(query) abort
-    if type(a:query) isnot v:t_string || empty(a:query)
+function! ale#symbol#Search(args) abort
+    if type(a:args) isnot v:t_string || empty(a:args)
```
[the remark this answers is not in the document]
Comment (chaucerbao) [reply autoload/ale/symbol.vim]: Done

**Comment by chaucerbao** [comment]
@w0rp What's the best way to request another review after change requests are addressed?

**Comment by w0rp** [comment]
Check out 19cc724807de4cdf9f8147c70932b49a79a7f7ee and examples in `test/test_parse_command_args.vader`. I have now added a function for parsing command arguments we can use everywhere.

**Comment by w0rp** [comment]
If you wait, I'll look at the pull request eventually.

**Comment by chaucerbao** [comment]
Gotcha, didn't mean to nag. It just isn't clear what state the PR is in after a change request.

**Pull request merged**